test_views.py

```
@@ -0,0 +1,315 @@
+            return JsonResponse({'error': 'Product ID required'}, status=400)
+        
+        product = get_object_or_404(Product, id=product_id)
+        product.name = data.get('name', product.name)
+        product.full_clean()  # Validate the model
+        product.save()
+        return JsonResponse({'status': 'success'})
+    except json.JSONDecodeError:
+        return JsonResponse({'error': 'Invalid JSON'}, status=400)
+    except ValidationError as e:
+        return JsonResponse({'error': str(e)}, status=400)
+    except Exception as e:
```

Ruff (0.14.10)

Do not catch blind exception: `Exception`

(BLE001)